docs/develop/python/message-passing.mdx

```diff
@@ -413,131 +499,120 @@ class MyWorkflow:
-
-1. **Workflow finished while the Update handler execution was in progress**
-
-    You'll get a [`temporalio.service.RPCError`](https://python.temporal.io/temporalio.service.RPCError.html) "workflow execution already completed"`
-
-    This will happen if the Workflow finished while the Update handler execution was in progress, for example because
     - The Workflow was canceled or failed.
-    - The Workflow completed normally or continued-as-new and the Workflow author did not [wait for handlers to be finished](/encyclopedia/workflow-message-passing#finishing-message-handlers).
 
-### Problems when sending a Query
-
-1. **There is no Workflow Worker polling the Task Queue**
+    - The Workflow completed normally or continued-as-new and the Workflow author did not [wait for handlers to be finished](/encyclopedia/workflow-message-passing#finishing-message-handlers).
 
-    You'll get an [`temporalio.service.RPCError`](https://python.temporal.io/temporalio.service.RPCError.html) on which the `status` attribute is [`RPCStatusCode`](https://python.temporal.io/temporalio.service.RPCStatusCode.html) `FAILED_PRECONDITION`.
+### Problems when sending a Query {#query-problems}
 
-1. **Query failed.**
+When working with Queries, you may encounter these errors:
 
-    You'll get [`temporalio.client.WorkflowQueryFailedError`](https://python.temporal.io/temporalio.client.WorkflowQueryFailedError.html).
-    Any exception in a Query handler will cause this.
-    Note that this differs from Signal and Update, for which exceptions can cause Workflow Task Failure.
+- **There is no Workflow Worker polling the Task Queue**:
+    You'll receive a [`temporalio.service.RPCError`](https://python.temporal.io/temporalio.service.RPCError.html) exception on which the `status` attribute is [`RPCStatusCode`](https://python.temporal.io/temporalio.service.RPCStatusCode.html) `FAILED_PRECONDITION`.
 
-1. **The handler caused the Workflow Task to fail.**
+- **Query failed**:
+  You'll receive a [`temporalio.client.WorkflowQueryFailedError`](https://python.temporal.io/temporalio.client.WorkflowQueryFailedError.html) exception if something goes wrong during a Query.
+  Any exception in a Query handler will trigger this error.
+  This differs from Signal and Update requests, where exceptions can lead to Workflow Task Failure instead.
 
+- **The handler caused the Workflow Task to fail.**
     This would happen, for example, if the Query handler blocks the thread for too long without yielding.
 
+## Dynamic components {#dynamic-handler}
 
-## Dynamic Handlers {#dynamic-handler}
+A dynamic Workflow, Activitity, Signal, Update, or Query is a kind of unnamed item.
+Normally, these items are registered by name with the Worker and invoked at runtime.
+When an unregistered or unrecognized Workflow, Activity, or message handler arrives with a recognized method signature, the Worker can use a pre-registered dynamic stand-in.
```

Comment:
```suggestion
When an unregistered or unrecognized Workflow, Activity, or message request arrives with a recognized method signature, the Worker can use a pre-registered dynamic stand-in.
```

Reply:
"handler" was wrong. Suggesting "request" there since we don't normally speak of Workflows or Activities "arriving".

Reply:
Adopted